apps/src/netsim/NetSimVisualization.js

```diff
@@ -322,7 +303,6 @@ NetSimVisualization.prototype.onNodeTableChange_ = function (rows) {
   // Update collection of VizNodes from source data
   this.updateVizEntitiesOfType_(NetSimVizSimulationNode, tableNodes, function (node) {
     var newVizNode = new NetSimVizSimulationNode(node);
-    newVizNode.setDnsMode(this.netsim_.getDnsMode());
```

Comment:
I've added some tests on your branch for the behavior of `setDnsMode` and `setEncodings` that cover this case.